src/test/java/com/microsoft/azure/relay/SendReceiveTest.java

```diff
@@ -204,6 +204,24 @@ public void httpLargePOSTAndLargeResponseTest() throws IOException {
         httpRequestSender("POST", largeStr, largeStr);
 	}
 	
+	@Test
+	public void httpWriteSmallThenSmallResponseTest() throws IOException {
```

Comment:
>httpWriteSmallThenSmallResponseTest [](start = 13, length = 35)

Would anything in the tests catch and fail on a scenario where the message arrived with extra '\0' characters? (Converting from bytes to String might throw away trailing '\0' chars...

"FirstSecond\0\0\0\0"